sql/hive/src/main/java/org/apache/spark/sql/hive/mapred/CombineSplitInputFormat.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.hive.mapred;
+
+import java.io.IOException;
+import java.util.*;
+
+import com.clearspring.analytics.util.Lists;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+import org.apache.hadoop.mapred.*;
+
+public class CombineSplitInputFormat<K, V> implements InputFormat<K, V> {
+
+  private InputFormat<K, V> inputformat;
+  private long splitSize = 0;
+
+  public CombineSplitInputFormat(InputFormat<K, V> inputformat, long splitSize) {
+    this.inputformat = inputformat;
+    this.splitSize = splitSize;
+  }
+
+  /**
+   * Create a single split from the list of blocks specified in validBlocks
+   * Add this new split into splitList.
+   */
+  private void addCreatedSplit(List<CombineSplit> splitList,
+                               long totalLen,
+                               Collection<String> locations,
+                               List<InputSplit> validSplits) {
+    CombineSplit combineSparkSplit =
+      new CombineSplit(validSplits.toArray(new InputSplit[0]),
+        totalLen, locations.toArray(new String[0]));
```

Comment:
Leave a TODO, we probably can optimize this by providing the second/third optimal locations.